torch/distributed/tensor/_ops/_tensor_ops.py

```diff
@@ -741,7 +741,7 @@ def prop_index_select(op_schema: OpSchema) -> OutputSharding:
 def prop_index_put(op_schema: OpSchema) -> StrategyType:
     # We have 3 DTensor spec from argument `in`, `indices` and `values`
     # accordingly.
-    in_spec, indices_spec, values_spec = op_schema.args_schema
+    in_spec, indices_spec, values_spec, *_ = op_schema.args_schema
```

Comment:
I am wondering if this sharding prop still misses some valid cases, though they might not be very common.

If 'self' is partial, I think there are some new options open

| self  | index | values | accumulate |
| ------------- | ------------- | --- | --- |
|Partial|Replicate| Replicate|True|
|Partial|Replicate| Shard|True|
|Partial|Shard| Shard|True|